src/connectors/paymentService.ts

```diff
@@ -760,5 +786,50 @@ export class PaymentService extends BaseService {
         userId,
       })
       .returning('*')
+
+    // Mark coupon invitation as accepted
+    if (ivt) {
+      await this.acceptCouponInvitation(ivt.id)
+    }
+  }
+
+  /*********************************
+   *                               *
+   *            Coupon             *
+   *                               *
+   *********************************/
+  /**
+   * Find coupons applicable to a user for a cirlce
+   */
+  findPendingInvitation = async (params: {
+    userId: string
+    priceId: string
+  }) => {
+    const records = await this.knex
+      .select('ci.id', 'cc.provider_coupon_id')
+      .from('circle_invitation as ci')
+      .join('circle_price as cp', 'cp.circle_id', 'ci.circle_id')
+      .join('circle_coupon as cc', 'cc.id', 'ci.coupon_id')
+      .where({
```

Comment:
Since we allow circle owners to invite via email, we should find invitation via `user.email` also?

Reply:
merged for the time being, I will have another PR for this.

Reply:
@robertu7 please see the updated where clause for email checking

Reply:
Nice point. 💪🏻